src/api/rate_limiter.py

```diff
@@ -78,31 +80,81 @@ async def handle_requests(self):
-                    self.logger.warning(
-                        "Request : GET %s + %s failed with response code %s",
+                try:
+                    resp = requests.get(request.url, cookies=request.cookies, timeout=DEFAULT_MAX_TIMEOUT)
+                except requests.exceptions.Timeout:
+                    self.logger.error("Request GET %s + %s: the request timed out", 
                         request.url,
                         request.cookies,
-                        resp.status_code,
+                        retry_count, 
+                        self._max_retry
                     )
                     if retry_count < self._max_retry:
                         retry = True
                         retry_count += 1
-                    else:
+                        continue
+                    
+                    self.logger.error("Request timed out too many times. Potential ban")
+                    
+                    # TODO Send message on discord
+                    
+                    self.logger.error("Waiting 10min.")
+                    await asyncio.sleep(600)
+                    self.logger.warning("Waking up: ready to retry after timeout")
+                    retry = True
+                    retry_count += 1
+                    continue
+
+                if resp.status_code == 429:
+                    try:
+                        wait = int(resp.headers['Retry-After'])
+                    except:
                         self.logger.error(
-                            "Failed to get request after %s attempt. We could be banned :(",
-                            self._max_retry,
+                            "Request got 429 and failed to parse headers: %s",
+                            resp.headers
                         )
-                        raise RuntimeError("Looks like a ban to me :'(")
-                else:
-                    data = resp.json()
+                        raise RuntimeError("Wrong headers")
+                    
+                    self.logger.warning(
+                        "API overload (429) with request : GET %s + %s. Waiting %ssec.",
+                        request.url,
+                        request.cookies,
+                        wait,
+                    )
+                    await asyncio.sleep(wait)
+                    self.logger.warning("Waking up: ready to retry after 429")
+                    retry = True
+                    retry_count += 1
+                    continue
+
+                elif resp.status_code != 200:
+                    self.logger.warning(
+                        "Request : GET %s + %s failed with response code %s. Try: %s/%s",
+                        request.url,
+                        request.cookies,
+                        resp.status_code,
+                    )
+                    if retry_count < self._max_retry:
+                        # Retry the request
+                        retry = True
```

Comment:
I've been thinking and I'm pretty sure this `retry` variable is useless... the only time it's used we could use `retry_count` I think